File: keevo/app/lib/features/catalog/domain/model/product_response_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product_response_dto.freezed.dart';
part 'product_response_dto.g.dart';

/// ProductResponseDto — backend API response for a product.
///
/// Returned by GET /api/v1/products and POST /api/v1/products.
/// Maps directly from the backend [ProductResponseDto] JSON structure.
@freezed
class ProductResponseDto with _$ProductResponseDto {
  const factory ProductResponseDto({
    required String id,
    required String name,
    String? description,
    required String sku,
    String? categoryId,
    @Default(0) int price,
    @Default(0) int buyPrice,
    @Default(0) int stockQuantity,
    String? photoUrl,
    @Default(false) bool archived,
    @Default('ACTIVE') String status,
    required String createdAt,
    required String updatedAt,
  }) = _ProductResponseDto;

  factory ProductResponseDto.fromJson(Map<String, dynamic> json) =>
      _$ProductResponseDtoFromJson(json);
}

File: keevo/backend/src/main/java/com/keevo/catalog/product/adapter/in/web/dto/ProductResponseDto.java
package com.keevo.catalog.product.adapter.in.web.dto;

import com.keevo.catalog.product.domain.entity.Product;
import com.keevo.catalog.product.domain.entity.ProductStatus;

import java.time.Instant;
import java.util.UUID;

/**
 * Response DTO for Product API endpoints
 * 
 * Includes all product information for frontend consumption
 */
public record ProductResponseDto(
        UUID id,
        String name,
        String description,
        String sku,
        UUID categoryId,
        Integer price,
        Integer buyPrice,
        Integer transportCost,
        Integer stockQuantity,
        boolean archived,
        ProductStatus status,
        Instant createdAt,
        Instant updatedAt
) {
    /**
     * Factory method to create DTO from domain entity
     */
    public static ProductResponseDto fromDomain(Product product) {
        return new ProductResponseDto(
                product.getId(),
                product.getName(),
                product.getDescription(),
                product.getSku(),
                product.getCategoryId(),
                product.getPriceValue(),
                product.getBuyPriceValue(),
                product.getTransportCostValue(),
                product.getStockQuantity(),
                product.getArchived(),
                product.getStatus(),
                product.getCreatedAt(),
                product.getUpdatedAt()
        );
    }
}